pkg/controller/cainjector/setup.go

```diff
@@ -21,143 +21,135 @@ import (
 	"k8s.io/apimachinery/pkg/runtime"
 	apireg "k8s.io/kube-aggregator/pkg/apis/apiregistration/v1"
 	ctrl "sigs.k8s.io/controller-runtime"
-	"sigs.k8s.io/controller-runtime/pkg/cache"
 	"sigs.k8s.io/controller-runtime/pkg/client"
-	"sigs.k8s.io/controller-runtime/pkg/cluster"
-	"sigs.k8s.io/controller-runtime/pkg/controller"
 	"sigs.k8s.io/controller-runtime/pkg/handler"
-	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	cmapi "github.com/cert-manager/cert-manager/pkg/apis/certmanager/v1"
 )
 
 // injectorSet describes a particular setup of the injector controller
 type injectorSetup struct {
 	resourceName string
 	injector     CertInjector
 	listType     runtime.Object
+	objType      client.Object
 }
 
 var (
 	MutatingWebhookSetup = injectorSetup{
 		resourceName: "mutatingwebhookconfiguration",
 		injector:     mutatingWebhookInjector{},
 		listType:     &admissionreg.MutatingWebhookConfigurationList{},
+		objType:      &admissionreg.MutatingWebhookConfiguration{},
 	}
 
 	ValidatingWebhookSetup = injectorSetup{
 		resourceName: "validatingwebhookconfiguration",
 		injector:     validatingWebhookInjector{},
 		listType:     &admissionreg.ValidatingWebhookConfigurationList{},
+		objType:      &admissionreg.ValidatingWebhookConfiguration{},
 	}
 
 	APIServiceSetup = injectorSetup{
 		resourceName: "apiservice",
 		injector:     apiServiceInjector{},
 		listType:     &apireg.APIServiceList{},
+		objType:      &apireg.APIService{},
 	}
 
 	CRDSetup = injectorSetup{
 		resourceName: "customresourcedefinition",
 		injector:     crdConversionInjector{},
 		listType:     &apiext.CustomResourceDefinitionList{},
+		objType:      &apiext.CustomResourceDefinition{},
 	}
 
-	injectorSetups  = []injectorSetup{MutatingWebhookSetup, ValidatingWebhookSetup, APIServiceSetup, CRDSetup}
-	ControllerNames []string
+	injectorSetups = []injectorSetup{MutatingWebhookSetup, ValidatingWebhookSetup, APIServiceSetup, CRDSetup}
 )
 
 // registerAllInjectors registers all injectors and based on the
 // graduation state of the injector decides how to log no kind/resource match errors
-func registerAllInjectors(ctx context.Context, groupName string, mgr ctrl.Manager, sources []caDataSource, client client.Client, ca cache.Cache, namespace string) error {
-	controllers := make([]controller.Controller, len(injectorSetups))
-	for i, setup := range injectorSetups {
-		controller, err := newGenericInjectionController(ctx, groupName, mgr, setup, sources, ca, client, namespace)
-		if err != nil {
-			if !meta.IsNoMatchError(err) || !setup.injector.IsAlpha() {
-				return err
-			}
-			ctrl.Log.V(logf.WarnLevel).Info("unable to register injector which is still in an alpha phase."+
-				" Enable the feature on the API server in order to use this injector",
-				"injector", setup.resourceName)
-		}
-		controllers[i] = controller
+func RegisterAllInjectors(ctx context.Context, mgr ctrl.Manager, namespace string) error {
```

Comment:
Code in this function is not new, it is just pulled in from other parts like [here](https://github.com/cert-manager/cert-manager/blob/master/pkg/controller/cainjector/setup.go#L119-L161) and `ApplyTo` methods of the different [data source implementations](https://github.com/cert-manager/cert-manager/blob/master/pkg/controller/cainjector/sources.go#L44-L60). I'd like to continue with refactor in another PR so that the change in this PR is possibly small